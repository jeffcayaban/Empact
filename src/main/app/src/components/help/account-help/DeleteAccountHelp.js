import React from "react";
import {Col, Grid, Panel, Row} from "react-bootstrap";
import {Link} from "react-router-dom";

import SettingsLink from "../../../other/images/settings-link.png";
import DeleteAccountScreen from "../../../other/images/delete-account-screen.png";
import DeleteConfirm from "../../../other/images/are-you-sure.png";
import SuccessfulAccountDeletion from "../../../other/images/successful-account-deletion.png";
import UnsuccessfulAccountDeletion from "../../../other/images/unsuccessful-account-deletion.png";
import BackButton from "../../common/BackButton";

/**
 * Displays help information for deleting an account.
 */

class DeleteAccountHelp extends React.PureComponent {

    render() {
        return (
            <div className={'secondaryPageContainerStyle'}>
                <div className={"centerStyle"}>
                    <h2>Deleting your Account</h2>
                    <p>Understand how to create delete your account.</p>
                </div>
                <div className={"titleContentPadding"}>
                    <Grid className={"divNoPadding"}>
                        <Row>
                            <Col md={12} lg={12}>
                                <div>
                                    <h4 style={{ margin: 0 }}>How to delete your account</h4>
                                    <hr className={"helpHr"} />

                                    <p>
                                        1. Go to the <Link to={"/settings"}>settings page</Link>. The link is found on
                                        the top navigation bar.
                                    </p>

                                    <div className={"centerStyle"} style={{ paddingBottom: '15px' }}>
                                        <img style={{ maxWidth: '25%', minWidth: '350px', align: 'center' }}
                                             src={SettingsLink}
                                             alt="Settings Link"
                                        />
                                    </div>

                                    <p>
                                        2. Click on "Delete Account" link and then click on the "Delete Account" button.
                                    </p>

                                    <div className={"centerStyle"} style={{ paddingBottom: '20px' }}>
                                        <img style={{ maxWidth: '25%', minWidth: '350px', align: 'center' }}
                                             src={DeleteAccountScreen}
                                             alt="Delete Account Screen"
                                        />
                                    </div>

                                    <p>
                                        3. A dialog will be displayed to confirm whether you want to delete your account
                                        where clicking on "Yes" will delete the account. Below shows an image of the
                                        expected confirmation dialog.
                                    </p>

                                    <div className={"centerStyle"} style={{ paddingTop: '20px', paddingBottom: '20px' }}>
                                        <img style={{ maxWidth: '30%', minWidth: '350px', align: 'center' }}
                                             src={DeleteConfirm}
                                             alt="Example of a delete confirmation dialog"
                                        />
                                    </div>


                                    <p>
                                        4. If the deletion is successful then you will be shown a confirmation screen,
                                        otherwise an error screen will be displayed.
                                    </p>

                                    <div className={"centerStyle"} style={{ paddingTop: '20px' }}>
                                        <Grid className={"divNoPadding"}>
                                            <Row>
                                                <Col md={6} lg={6}>
                                                    <Panel>
                                                        <Panel.Heading>
                                                            <Panel.Title>
                                                                If successful
                                                            </Panel.Title>
                                                        </Panel.Heading>
                                                        <Panel.Body className={"centerStyle"}>
                                                            <img style={{ width: '320px', align: 'center' }}
                                                                 src={SuccessfulAccountDeletion}
                                                                 alt="Successful Account Deletion Screen"
                                                            />
                                                        </Panel.Body>
                                                    </Panel>
                                                </Col>
                                                <Col md={6} lg={6}>
                                                    <Panel>
                                                        <Panel.Heading>
                                                            <Panel.Title>
                                                                If unsuccessful
                                                            </Panel.Title>
                                                        </Panel.Heading>
                                                        <Panel.Body className={"centerStyle"}>
                                                            <img style={{ width: '320px', align: 'center' }}
                                                                 src={UnsuccessfulAccountDeletion}
                                                                 alt="Unsuccessful Account Deletion Screen"
                                                            />
                                                        </Panel.Body>
                                                    </Panel>
                                                </Col>
                                            </Row>
                                        </Grid>
                                    </div>
                                </div>
                                <BackButton link={"/help/account"}/>
                            </Col>
                        </Row>
                    </Grid>
                </div>
            </div>
        );
    }
}

export default DeleteAccountHelp;