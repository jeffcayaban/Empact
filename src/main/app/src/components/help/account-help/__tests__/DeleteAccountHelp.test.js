import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import DeleteAccountHelp from "../DeleteAccountHelp";

describe('DeleteAccountHelp', () => {

    it('should render successfully', () => {
        const component = shallow(<DeleteAccountHelp />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
