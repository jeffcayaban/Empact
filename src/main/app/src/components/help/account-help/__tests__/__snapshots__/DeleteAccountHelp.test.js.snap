// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAccountHelp should render successfully 1`] = `
<div
  className="secondaryPageContainerStyle"
>
  <div
    className="centerStyle"
  >
    <h2>
      Deleting your Account
    </h2>
    <p>
      Understand how to create delete your account.
    </p>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          lg={12}
          md={12}
        >
          <div>
            <h4
              style={
                Object {
                  "margin": 0,
                }
              }
            >
              How to delete your account
            </h4>
            <hr
              className="helpHr"
            />
            <p>
              1. Go to the 
              <Link
                replace={false}
                to="/settings"
              >
                settings page
              </Link>
              . The link is found on the top navigation bar.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingBottom": "15px",
                }
              }
            >
              <img
                alt="Settings Link"
                src="settings-link.png"
                style={
                  Object {
                    "align": "center",
                    "maxWidth": "25%",
                    "minWidth": "350px",
                  }
                }
              />
            </div>
            <p>
              2. Click on "Delete Account" link and then click on the "Delete Account" button.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingBottom": "20px",
                }
              }
            >
              <img
                alt="Delete Account Screen"
                src="delete-account-screen.png"
                style={
                  Object {
                    "align": "center",
                    "maxWidth": "25%",
                    "minWidth": "350px",
                  }
                }
              />
            </div>
            <p>
              3. A dialog will be displayed to confirm whether you want to delete your account where clicking on "Yes" will delete the account. Below shows an image of the expected confirmation dialog.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingBottom": "20px",
                  "paddingTop": "20px",
                }
              }
            >
              <img
                alt="Example of a delete confirmation dialog"
                src="are-you-sure.png"
                style={
                  Object {
                    "align": "center",
                    "maxWidth": "30%",
                    "minWidth": "350px",
                  }
                }
              />
            </div>
            <p>
              4. If the deletion is successful then you will be shown a confirmation screen, otherwise an error screen will be displayed.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingTop": "20px",
                }
              }
            >
              <Grid
                bsClass="container"
                className="divNoPadding"
                componentClass="div"
                fluid={false}
              >
                <Row
                  bsClass="row"
                  componentClass="div"
                >
                  <Col
                    bsClass="col"
                    componentClass="div"
                    lg={6}
                    md={6}
                  >
                    <Uncontrolled(Panel)>
                      <PanelHeading
                        bsClass="panel"
                        componentClass="div"
                      >
                        <PanelTitle
                          bsClass="panel"
                          componentClass="div"
                        >
                          If successful
                        </PanelTitle>
                      </PanelHeading>
                      <PanelBody
                        bsClass="panel"
                        className="centerStyle"
                        collapsible={false}
                      >
                        <img
                          alt="Successful Account Deletion Screen"
                          src="successful-account-deletion.png"
                          style={
                            Object {
                              "align": "center",
                              "width": "320px",
                            }
                          }
                        />
                      </PanelBody>
                    </Uncontrolled(Panel)>
                  </Col>
                  <Col
                    bsClass="col"
                    componentClass="div"
                    lg={6}
                    md={6}
                  >
                    <Uncontrolled(Panel)>
                      <PanelHeading
                        bsClass="panel"
                        componentClass="div"
                      >
                        <PanelTitle
                          bsClass="panel"
                          componentClass="div"
                        >
                          If unsuccessful
                        </PanelTitle>
                      </PanelHeading>
                      <PanelBody
                        bsClass="panel"
                        className="centerStyle"
                        collapsible={false}
                      >
                        <img
                          alt="Unsuccessful Account Deletion Screen"
                          src="unsuccessful-account-deletion.png"
                          style={
                            Object {
                              "align": "center",
                              "width": "320px",
                            }
                          }
                        />
                      </PanelBody>
                    </Uncontrolled(Panel)>
                  </Col>
                </Row>
              </Grid>
            </div>
          </div>
          <BackButton
            link="/help/account"
          />
        </Col>
      </Row>
    </Grid>
  </div>
</div>
`;
